linera-chain/src/data_types.rs

```diff
@@ -916,7 +916,7 @@ pub(crate) fn check_signatures(
     );
     // All that is left is checking signatures!
     let hash_and_round = VoteValue(value_hash, round, certificate_kind);
-    ValidatorSignature::verify_batch(&hash_and_round, signatures.iter().map(|(v, s)| (&v.0, s)))?;
+    ValidatorSignature::verify_batch(&hash_and_round, signatures.iter().map(|(v, s)| (v, s)))?;
```

Comment:
```suggestion
    ValidatorSignature::verify_batch(&hash_and_round, &signatures)?;
```

Reply:
This won't work. The argument type is an iterator over references - not a reference to something that can be turned into an iterator.

Reply:
`verify_batch` already accepts `IntoIterator`.